src/storage/compression/dictionary/decompression.cpp

```diff
@@ -109,6 +116,8 @@ void CompressedStringScanState::ScanToDictionaryVector(ColumnSegment &segment, V
 	BitpackingPrimitives::UnPackBuffer<sel_t>(dst, src, scan_count, current_width);
 
 	result.Dictionary(*(dictionary), dictionary_size, *sel_vec, scan_count);
+	// FIXME: this assumes the type is VectorType::DICTIONARY
+	// this could fail if the result is a ConstantVector instead.
```

Comment:
I don't really get this fixme, result is explicitly set to a dictionary one line above this one?

Reply:
Yea but that can result in a flat vector

Reply:
It's probably not realistic here, but the Dictionary method looks a little broken
when called on a constant it remains a Constant, when called on an FSST vector it becomes Flat

```c++
	if (GetVectorType() == VectorType::FSST_VECTOR) {
		Flatten(sel, count);
		return;
	}

	Vector child_vector(*this);
	auto internal_type = GetType().InternalType();
	if (internal_type == PhysicalType::STRUCT) {
		child_vector.auxiliary = make_buffer<VectorStructBuffer>(*this, sel, count);
	}
	auto child_ref = make_buffer<VectorChildBuffer>(std::move(child_vector));
	auto dict_buffer = make_buffer<DictionaryBuffer>(sel);
	vector_type = VectorType::DICTIONARY_VECTOR;
	buffer = std::move(dict_buffer);
	auxiliary = std::move(child_ref);
```